test/tag_test.dart

```diff
@@ -6,31 +6,39 @@ import 'test_utils.dart';
 void main() {
   test('Parse lightweight tag', () async {
     final testDir = await createTempGitDir();
-
     final contents = <String, String>{'something': 'value'};
+    const givenTagName = 'newTag';
 
     await doDescriptorGitCommit(testDir, contents, 'Something');
     final branchRef = await testDir.currentBranch();
 
     await runGit(
-      ['tag', 'newTag', branchRef.sha],
+      ['tag', givenTagName, branchRef.sha],
       processWorkingDir: testDir.path,
     );
 
-    final tagsFound = await testDir.tags().toList();
-    expect(tagsFound, hasLength(1));
+    final foundTag = (await testDir.tags().toList()).first;
```

Comment:
```suggestion
    final foundTag = (await testDir.tags().toList()).single;
```

nice way to ensure only one item